mautic/mautic
# Add custom validation message to international phone number form validation

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL |  https://github.com/mautic/developer-documentation/pull/109
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Based on quick fix https://github.com/mautic/mautic/pull/6645 and feature added to 2.14.1 https://github.com/mautic/mautic/pull/5400 we add option to change validation message for  nternational phone number form validation. 

This PR also add extend addValidator event with parameter formType, which allow extend validation tab by developers (see docs https://github.com/mautic/developer-documentation/pull/109)

This is example how developers can add options to validator tab and then validate based on setting: https://github.com/mautic/mautic/pull/6650/files#diff-e68b0c198b71013d6f317456841c2abd

![image](https://user-images.githubusercontent.com/462477/46532632-fcb7e480-c8a1-11e8-8efd-8f385a4e95e3.png)


#### Steps to test this PR:
1: Create form
2. Create Phone field. See properties tab and option: Validate as internationally-standardized format. Enable it and also setup validation message
3. Try send form with non validated form nubmer (xxx for example)
4. You should see your custom message
5. Then go to phone field. And remove custom validation text message
6. Try send form with non validated form nubmer (xxx for example)
7. You should see default validation message
